src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +265,116 @@ namespace AppInstaller::CLI::Workflow
+
+        if (enr.Status() == GetEntitlementStatus::Succeeded)
+        {
+            context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
+            AICLI_LOG(CLI, Error, << "Get entitlement succeeded.");
+        }
+        else
+        {
+            if (enr.Status() == GetEntitlementStatus::NoStoreAccount)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNoStoreAccount << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. No Store account.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::NetworkError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNetworkError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. Network error.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::ServerError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementServerError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement succeeded. Server error. ProductId: " << Utility::ConvertToUTF8(productId));
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_INSTALL_FAILED);
+        }
+
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        IVectorView<AppInstallItem> installItems = installManager.StartProductInstallAsync(
+            productId,              // ProductId
+            winrt::hstring(),       // CatalogId
+            winrt::hstring(),       // FlightId
+            L"WinGetCli",           // ClientId
+            false,                  // repair
+            false,
+            winrt::hstring(),
+            nullptr).get();
+
+        // We install one package at a time so the AppInstallItem list returned should contain only one entry.
+        auto installItem = installItems.GetAt(0);
+
+        auto CheckInstallStatus = [&]()
+        {
+            HRESULT errorCode = installItem.GetCurrentStatus().ErrorCode();
+            if (!SUCCEEDED(errorCode))
+            {
+                // Cancel the installation request on failure. Otherwise it'll leave an error
+                // request in Store client installation queue.
+                installItem.Cancel();
+
+                context.Reporter.Info() << Resource::String::MSStoreInstallFailed << ' ' << errorCode << std::endl;
+                AICLI_LOG(CLI, Error, << "MSStore install failed. ProductId: " << Utility::ConvertToUTF8(productId) << " HResult: " << errorCode);
+                AICLI_TERMINATE_CONTEXT(errorCode);
+            }
+        };
+
+        context.Reporter.ExecuteWithProgress(
+            [&](IProgressCallback& progress)
+            {
+                while (installItem.GetCurrentStatus().PercentComplete() < 100)

```

Comment:
>installItem.GetCurrentStatus().PercentComplete() [](start = 23, length = 48)

Since you need this 3 times each iteration and it might require a cross process COM call to get, maybe just save it in a local each time through the loop?